### ISSUE TITLE
Allow hostpubmd5 argument  to curl / sftp

### ISSUE THREAD

**Issue by camlafit** [git-ftp/git-ftp#267]
Hi

I don't understand where is the problem but with sftp command, curl check in know_host file fingerprint. To do this check a md5 fingerprint search is done. Problem occurs because this file has only public key and not fingerprint.
To solve this problem we have two choice : 
* add directly this fingerprint in know_host , ugly solution
* use --hostpubmd5 curl argument to set the fingerprint waited.

Is it possible to add this new argument in config file ?

Tks

**Comment by mkllnk** [comment]
You are right. Git-ftp doesn't support that argument at the moment. Since there are lots of curl arguments, it would probably be good if you could pass any argument to curl.

**Comment by camlafit** [comment]
Hi

In this case maybe provide a generic prefix to distingue git ftp argument
and specific curl ?
Would be easier to detect these arguments and pass theses to curl function

**Comment by ThomasBlt** [comment]
`--hostpubmd5` is SSH counterpart to TLS's `--cacert` so why support only TLS and force sftp users to resort to enable `--insecure`.

MD5 is not safe anymore **it's a BAD IDEA to use this option** ... yet it's curl sole option 

If you wan't to handle this problem the correct way you just should `ssh YOUR_SFTP_TARGET` and you'll be prompted with the usual

If, as me, your experiencing this issue on windows/cygwin/sourcetree the problem might be that curl is searching `~/.ssh/known_hosts` under `~/_ssh/known_hosts` **==> SOLUTION** `ln -s .ssh/ _ssh` in your home dir

(BTW [git-ftp.1.md#netrc](https://github.com/git-ftp/git-ftp/blob/master/man/git-ftp.1.md#netrc) won't help here because it's focused on FTP ...)

**Comment by andypillip** [comment]
Moin.

NETRC will work for any protocol with curl.
>  curl does not, however, limit this feature to FTP, but can get credentials for machines for any protocol with this.
